src/app/accounts/add-wallet.ts

```diff
@@ -0,0 +1,23 @@
+import { AccountsRepository, WalletsRepository } from "@services/mongoose"
+
+export const addWallet = async ({
+  accountId,
+  type,
+  currency,
+}: {
+  accountId: AccountId
+  type: WalletType
+  currency: WalletCurrency
+}): Promise<Wallet | ApplicationError> => {
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof Error) return account
+
+  const wallet = await WalletsRepository().persistNew({
+    accountId,
+    type,
+    currency,
+  })
+  if (wallet instanceof Error) return wallet
+
+  return wallet
```

Comment:
I could be tempted to remove the `account = ...` line in a future update if I didn't understand that we'd like to check that the account exists first. This could either be a good place for a code comment, or we could alternatively structure code like below.

We should also add a test calling this function with an `accountId` that doesn't exist.

```suggestion
  return !(account instanceof Error)
    ? WalletsRepository().persistNew({
        accountId,
        type,
        currency,
      })
    : account
```

Reply:
moved it to the persistNew in WalletsRepository